### DISCUSSION TITLE
Passing variables into the *.mmt file

### DISCUSSION THREAD

**Discussion by anastasia-rk** [MichaelClerx/myokit#1031] [Q&A] [answered]
Hi Michael! 

I finally got around to use myokit and have a newbie question about passing values into the *.mmt file. If I use  `simulation.set_constant()` method in the embedded script to assign a conductance value, do I also need to use `bind`  for the corresponding conductance variable within the model file itself? I am currently doing both but can’t help thinking that it is an overkill. Many thanks in advance!

**Comment by MichaelClerx** [comment, marked as answer]
Hi anastasia!

The bind keyword is used to tie *time-varying simulator inputs* to model variables, the main examples being `time` for the time variable,  `pace` for the voltage clamp or stimulus protocol, and `diffusion_current` for the cell-to-cell current in multi-cell simulations. So you don't need it for constants!

Hope that helps,
  Michael

Bit more background: By using the `bind` keyword to label a variable, you are declaring it as a placeholder that a simulation engine may replace with an appropriate input. So writing `t = 0 [ms] bind time` creates a variable that will have a value set by the simulation if the label `time` corresponds to a supported input, and `0 [ms]` if not. For this reason, the `Model` will not consider bound variables to be constants: calling [is_constant](https://myokit.readthedocs.io/en/stable/api_core/Variable.html#myokit.Variable.is_constant) on a bound variable returns False. However... in the `Simulation` any unsupported bindings will fall back on the right-hand side provided by the model, which is why `set_constant` still works in your case.
